src/elements/content-sidebar/MetadataSidebarRedesign.tsx

```diff
@@ -1,32 +1,124 @@
+function MetadataSidebarRedesign({ api, fileId, isFeatureEnabled, onError, selectedTemplateKey, templateFilters }) {
+    const [editors, setEditors] = useState<Array<MetadataEditor>>([]);
+    const [error, setError] = useState<MessageDescriptor>(undefined);
+    const [file, setFile] = useState<BoxItem>(undefined);
+    const [templates, setTemplates] = useState<Array<MetadataTemplate>>([]);
+
+    const onApiError = (err: ElementsXhrError, code: string) => {
+        const { status } = err;
+        const isValidError = isUserCorrectableError(status);
+        setError(messages.sidebarMetadataEditingErrorContent);
+        onError(error, code, { error, [IS_ERROR_DISPLAYED]: isValidError });
+    };
+
+    const fetchMetadataErrorCallback = (e: ElementsXhrError, code: string) => {
+        onApiError(e, code);
+
+        setEditors(undefined);
+        setError(messages.sidebarMetadataFetchingErrorContent);
+        setTemplates(undefined);
+
+        // onApiError(e, code);
+        // MetadataSidebar.js
+        // onApiError(e, code, {
+        //     editors: undefined,
+        //     error: messages.sidebarMetadataFetchingErrorContent,
+        //     templates: undefined,
+        // });
+    };
+
+    const fetchMetadataSuccessCallback = ({ fetchedEditors, fetchedTemplates }) => {
+        setEditors(fetchedEditors ?? []); // fetchedEditors and fetchedTemplates are undefined now
+        setError(undefined);
+        setTemplates(normalizeTemplates(fetchedTemplates ?? [], selectedTemplateKey, templateFilters));
+    };
+
+    const fetchMetadata = fetchedFile => {
+        if (!fetchedFile) return;
+
+        api.getMetadataAPI(false).getMetadata(
+            fetchedFile,
+            fetchMetadataSuccessCallback,
+            fetchMetadataErrorCallback,
+            isFeatureEnabled,
+            { refreshCache: true },
+        );
+    };
+
+    const fetchFileErrorCallback = (e: ElementsXhrError, code: string) => {
+        onApiError(e, code);
+        setError(messages.sidebarFileFetchingErrorContent);
+        setFile(undefined);
+    };
+
+    const fetchFileSuccessCallback = (fetchedFile: BoxItem) => {
+        const currentCanUpload = getProp(file, FIELD_PERMISSIONS_CAN_UPLOAD, false);
+        const newCanUpload = getProp(fetchedFile, FIELD_PERMISSIONS_CAN_UPLOAD, false);
+        const shouldFetchMetadata = !file || currentCanUpload !== newCanUpload;
+        setFile(fetchedFile);
+        if (shouldFetchMetadata) fetchMetadata(fetchedFile);
+    };
+
+    const fetchFile = () => {
+        api.getFileAPI().getFile(fileId, fetchFileSuccessCallback, fetchFileErrorCallback, {
+            fields: [FIELD_IS_EXTERNALLY_OWNED, FIELD_PERMISSIONS],
+            refreshCache: true,
+        });
+    };
+
+    useEffect(() => {
+        fetchFile();
+    }, []);
+
+    const showEditor = !!file && !!templates && !!editors;
+    const showEmptyState = showEditor && editors && editors.length === 0;
```

Comment:
Do we need those `!!` in line 110? `!!` before `showEditor` in line 111 should produce the same result, however it looks like all those variables (in line 110) will be undefined in the 'error' scenario, so I believe there's no need for `!!` at all.